nav2_bringup/launch/manual_composed_bringup_launch.py

```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
```

Comment:
Where's the files that use this to actually bring up the full system for testing (simulation, etc)

Reply:
I could create a copy of `tb3_simulation_launch.py` , and just repalce  `bringup_launch.py` with `manual_composed_bringup_launch.py`, but it will cause a lot of repeated code. how about separating the part of launching Gazebo and spawning simulated robot from `tb3_simulation_launch.py` ?  @SteveMacenski

Reply:
All the configuration stuff would still be the same though and that is what takes up the bulk of the file anyway, so I don't think its worth abstracting something out unless we can abstract out more. What I don't want to have happen is just have a nest of launch files that are impossible for anyone to track (arguably this bringup system is already a bit complex due to the massive shift in thinking required to move to these files relative to ROS 1).

I'm debating whether we should have a new directory in launch for `composed/` or something that we have these 2 launch files in or if we should place them in the root `launch/` directory.

Reply:
i agree with you that make things easier. i think the best way to provide a flexible composed bringup system is `dynamically composed bringup` (based on ROS2 [Run-time composition](https://docs.ros.org/en/galactic/Tutorials/Composition.html)), which will be implemented in the future (maybe take a long time because of some problems: https://github.com/ros-planning/navigation2/issues/2147#issuecomment-915890304).

in my opinion, we needn't a general manually composed system if we have a dynamically composed system. the PR just provide a option/template for users to use `manual composition`. it's a simple example, perhaps we needn't consider choosing AmclNode or SLAM, let's consider these in `dynamically composed bringup`.

Reply:
how about adding a LaunchConfiguration `use_manually_composed_bringup` (or `use_manual_composition`), then `IfCondition` can be used to choose `bringup_launch.py` or `manual_composed_bringup_launch.py` for navigation?

Reply:
`use_composition` sounds good to me, no need to specify manual for a parameterization (too specific to the technique)